workers/data_refinery_workers/downloaders/sra.py

```diff
@@ -35,6 +36,15 @@ def _download_file(download_url: str,
         download_url = download_url.replace('.uk/', '.uk:/')
         return _download_file_aspera(download_url, downloader_job, target_file_path, source="ENA")
     elif "ncbi.nlm.nih.gov" in download_url and not force_ftp:
+        # Try to convert old-style endpoints into new-style endpoints if possible
+        try:
+            if 'anonftp' in download_url:
+                accession = download_url.split('/')[-1].split('.sra')[0]
+                new_url = get_fasp_sra_download(accession)
+                if new_url:
+                    download_url = new_url
```

Comment:
I think that we probably also need to update the OriginalFile object's `source_url` field because otherwise we won't actually know exactly where the file came from.

Reply:
Yeah I was thinking about that but I'm not sure if it's the right play as we destroy source information, for instance, if we want to fall back and use the old URL on retries, etc.

Reply:
Good point, and also I'm not convinced the URL we get back from NCBI is even persistent so storing it could be completely useless.